src/library/Dialog/__tests__/Dialog.spec.js

```diff
@@ -0,0 +1,211 @@
+  });
+
+  it('renders', () => {
+    const dialog = shallowDialog();
+
+    expect(dialog.exists()).toEqual(true);
+  });
+
+  describe('theme overrides', () => {
+    const props = {
+      id: 'test',
+      isOpen: true,
+      title: 'Test',
+      usePortal: false
+    };
+    testThemeOverrides(
+      <Dialog {...props} />,
+      getProcessedComponentThemeKeys(componentTheme, {
+        excludeKeys: ['Dialog_transitionDuration', 'DialogCloseButton_margin']
+      }).filter((key) => {
+        const parts = key.split('_');
+        const lastPart = parts[parts.length - 1];
+
+        return lastPart !== 'small' && lastPart !== 'large';
+      })
+    );
+    testThemeOverrides(<Dialog {...props} showCloseButton title="title" />, [
+      'DialogCloseButton_margin'
+    ]);
+  });
+
+  describe('opens', () => {
+    let wrapper;
+    let onOpen = jest.fn();
+
+    beforeEach(() => {
+      wrapper = mountInWrapper(
+        <Dialog title="test" usePortal={false} onOpen={onOpen} />
+      );
+
+      wrapper.setProps({ isOpen: true });
+    });
+
+    afterEach(() => {
+      onOpen.mockReset();
+    });
+
+    it('displays dialog content', () => {
+      const content = wrapper.find('DialogContent');
+
+      expect(content.exists()).toEqual(true);
+    });
+
+    it('focuses dialog root', () => {
+      const root = wrapper.find('[role="dialog"]').first();
+
+      expect(root.getDOMNode()).toEqual(document.activeElement);
+    });
+
+    it('calls onOpen callback', () => {
+      const dialog = wrapper.find(Dialog);
+
+      expect(dialog.props().onOpen).toHaveBeenCalled();
+    });
+  });
+
+  describe('closes', () => {
+    let trigger, wrapper;
+    let onClose = jest.fn();
+
+    beforeEach(() => {
+      trigger = global.document.createElement('button');
+      global.document.body.appendChild(trigger);
+      trigger.focus();
+
+      wrapper = mountInWrapper(
+        <Dialog title="test" usePortal={false} onClose={onClose} />
+      );
+
+      wrapper.setProps({ isOpen: true });
+    });
+
+    afterEach(() => {
+      onClose.mockReset();
+      global.document.body.removeChild(trigger);
+    });
+
+    it('when pressing Escape', () => {
+      const event = new KeyboardEvent('keydown', {
+        key: 'Escape',
+        view: window,
+        bubbles: true,
+        cancelable: true
+      });
+
+      document.body && document.body.dispatchEvent(event);
+      wrapper.update();
+
+      const content = wrapper.find('DialogContent');
+
+      expect(content.exists()).toEqual(false);
+    });
+
+    it('restores focus to trigger', () => {
+      wrapper.setProps({ isOpen: false });
+
+      expect(trigger).toEqual(document.activeElement);
+    });
+
+    xit('calls onClose callback', () => {
+      const dialog = wrapper.find(Dialog);
+
+      wrapper.setProps({ isOpen: false });
+      wrapper.update();
+
+      expect(dialog.props().onClose).toHaveBeenCalled();
+    });
+  });
```

Comment:
Are these (this & the test on line 181) ignored due to testing issues?

Reply:
Yes. 😢

Reply:
Yes. 😞